.github/workflows/code_coverage.yaml

```diff
@@ -0,0 +1,125 @@
+          composer bridge -- --coverage-clover bridge.xml
+          composer integration -- --coverage-clover integration.xml
+
+      - name: Upload Code Coverage to Codecov
+        run: |
+          curl -Os https://uploader.codecov.io/latest/linux/codecov
+          chmod +x codecov
+          ./codecov -t ${{ secrets.CODECOV_TOKEN }} -f component.xml -Z -F component
+          ./codecov -t ${{ secrets.CODECOV_TOKEN }} -f middleware.xml -Z -F middleware
+          ./codecov -t ${{ secrets.CODECOV_TOKEN }} -f bridge.xml -Z -F bridge
+          ./codecov -t ${{ secrets.CODECOV_TOKEN }} -f integration.xml -Z -F integration          
+
+  wordpress_tests_coverage:
+
+    name: WordPress Tests Coverage
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        php: [ 7.4 ]
+        wp: [ 5.9 ]
+        dependencies: [ highest ]
+
+    services:
+      mysql:
+        image: mysql:8.0.21
+        env:
+          MYSQL_ROOT_PASSWORD: root
+          MYSQL_DATABASE: sniccowp_1
+        ports:
+          - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval=3s --health-timeout=3s --health-retries=10
+
+      redis:
+        image: redis:5.0
+        ports:
+          - 6379:6379
+        options: --entrypoint redis-server
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP [${{ matrix.php }}]
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          tools: composer:v2, wp-cli
+          coverage: pcov
+          ini-values: error_reporting=E_ALL
+        env:
+          fail-fast: true
+
+      - name: Install composer dependencies
+        uses: "ramsey/composer-install@v1"
+        with:
+          dependency-versions: "${{ matrix.dependencies }}"
+
+      # Config value must match the ones in env.testing.dist
+      - name: Download WordPress [${{ matrix.wp }}]
+        run: |
+          mkdir wordpress && cd wordpress
+          wp core download --force --skip-content --version=${{ matrix.wp }}
+          wp config create --dbname="sniccowp_1" --dbuser="root" --dbpass="root" --dbhost="127.0.0.1" --dbprefix="wp_"
+          wp core install --url="sniccowp.test" --title="SniccoWP" --admin_user="admin" --admin_password="admin" --admin_email="admin@sniccowp.com" --skip-email
+          wp core update-db
+          wp rewrite structure '/%postname%/'
+          wp plugin install redis-cache
+          wp plugin activate redis-cache
+          wp redis enable
+        working-directory: /tmp
+
+      - name: Run WordPress tests with codeception
+        run: composer wordpress -- --coverage-xml wordpress.xml
+
+      - name: Upload Code Coverage to Codecov
+        run: |
+          curl -Os https://uploader.codecov.io/latest/linux/codecov
+          chmod +x codecov
+          ./codecov -t ${{ secrets.CODECOV_TOKEN }} -f wordpress.xml -Z -F wordpress
```

Comment:
WordPress flag makes no sense. We need a way to run codeception separately on each layer and then flag that aswell with either component, bridge or middleware